### PR TITLE
use custom runtime

### PR DESCRIPTION
**Pull request by elharo**
@nbashirbello @briandealwis Any idea what I'm doing wrong here? The classes are not found and as best I can tell that's because the runtimeClasspathProvider is never actually consulted. (I.e. debugger won't stop in its constructor)

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  @Override
+  public IClasspathEntry[] resolveClasspathContainer(IProject project, IRuntime runtime) {
+    return resolveClasspathContainer(runtime);
+  }
+
+  @Override
+  public IClasspathEntry[] resolveClasspathContainer(IRuntime runtime) {
+    String cloudSdkPath = PathResolver.INSTANCE.getCloudSdkPath().toString();
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java]: Remember that `PathResolver` may return `null` when no SDK is found.

**Comment by briandealwis** [comment]
> The classes are not found and as best I can tell that's because the runtimeClasspathProvider is never actually consulted. (I.e. debugger won't stop in its constructor)

I'm having _déja vu_…  Trying to remember…

**Comment by briandealwis** [comment]
You need to provide an adapter for the runtime-component to `IClasspathProvider`. The only provided adapter is an internal class `org.eclipse.jst.server.core.internal.RuntimeClasspathProvider$Factory`.

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java]: good catch. done.

**Comment by elharo** [comment]
PTAL @briandealwis 

Thanks. The adapter did the trick. This now successfully loads the classes from the custom runtime.

Next things to consider here:

1. Is there a non-internal factory class we should use in the adapter, or should we write our own?
2. How to improve the relative path resolution against the cloud SDK. As written I'm not confident this will work on Windows. 

Not sure if that should be part of this PR or the next batch of work.

**Comment by nbashirbello** [comment]
LGTM

**Pull request merged**